docs/src/modules/components/Notifications.js

```diff
@@ -89,25 +126,70 @@ export default function Notifications() {
     return () => {
       active = false;
     };
-  }, [route]);
+  }, []);
 
   return (
-    <Snackbar
-      key={message.id}
-      anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
-      ContentProps={{ 'aria-describedby': 'notification-message' }}
-      message={
-        <span id="notification-message" dangerouslySetInnerHTML={{ __html: message.text }} />
-      }
-      action={
-        <Button size="small" color="secondary" onClick={handleClose}>
-          {t('close')}
-        </Button>
-      }
-      open={open}
-      autoHideDuration={20e3}
-      onClose={handleClose}
-      onExited={handleMessage}
-    />
+    <React.Fragment>
+      <Tooltip
+        open={tooltipOpen}
+        onOpen={handleTooltipOpen}
+        onClose={handleTooltipClose}
+        title={t('notifications')}
+        enterDelay={300}
+      >
+        <IconButton
+          color="inherit"
+          ref={anchorRef}
+          aria-controls={open ? 'notifications-popup' : undefined}
+          aria-haspopup="true"
+          aria-label={t('notifications')}
+          onClick={handleToggle}
+          data-ga-event-category="AppBar"
+          data-ga-event-action="notifications"
+        >
+          <Badge color="secondary" badgeContent={unseenNotificationsCount}>
+            <NotificationsIcon />
+          </Badge>
+        </IconButton>
+      </Tooltip>
+      <NoSsr>
+        <Popper
+          id="notifications-popup"
+          anchorEl={anchorRef.current}
+          open={open}
+          placement="bottom-end"
+          transition
+          disablePortal
+        >
+          {({ TransitionProps }) => (
+            <Grow style={{ transformOrigin: 'top right' }} {...TransitionProps}>
```

Comment:
I couldn't get `transformOrigin` to work. Not sure what I'm missing?

Reply:
`TransitionProps` might have a `style` prop?

Reply:
Thanks, yes, I had wondered that too, but the order didn't make a difference.  Digging into the Popper source, it only sets `in`, `onEnter` and `onExited`. 😕

Reply:
Oh I think you just encountered https://github.com/facebook/react/issues/16393:

The component sets `transform` which is a shorthand-property while you set the `transformOrigin` longhand-property. React doesn't properly merge these. No idea how to fix these. Maybe try CSS classes?

Quite the timing since the issue resurfaced a few days ago.